File: 2000_ReversePrefixOfWord/reversePrefixOfWord.py
def reversePrefix(word: str, ch: str) -> str:
    i = j = 0
    arr = list(word)
    for char in word:
        if char == ch:
            while i < j:
                arr[i], arr[j] = arr[j], arr[i]
                i += 1
                j -= 1
            break
        j += 1
        
    return "".join(arr)

print(reversePrefix("abcdefd", "d"))
print(reversePrefix("xyxzxe", "z"))
print(reversePrefix("abcd", "z"))

File: 2000_ReversePrefixOfWord/reversePrefixOfWord.js
var reversePrefix = function(word, ch) {
    let i = 0
    let j = 0
    let arr = [...word]
    for (let k = 0; k < arr.length; k++) {
        if (arr[k] == ch) {
            while (i < j) {
                [arr[i], arr[j]] = [arr[j], arr[i]]
                i++
                j--
            }
            break
        }
        j++
    }
    return arr.join("")
};


console.log(reversePrefix("abcdefd", "d"))
console.log(reversePrefix("xyxzxe", "z"))
console.log(reversePrefix("abcd", "z"))